src/content/blog/summer-of-codegen.mdx

```diff
@@ -0,0 +1,440 @@
+
+```
+Use the current val name for the storage key, which you can get
+using the "pomdtr/extractValInfo" val, which you can call like this:
+\`\`\`ts
+const {author, name} = extractValInfo(import.meta.url);
+\`\`\`
+```
+
+### SQLite
+
+Each val has a [SQLite](https://docs.val.town/std/sqlite/) database, however LLMs had trouble with its API, since it uses arrays (instead of objects) as inputs and outputs. LLMs would consistently tripped up on this, so we told them to avoid SQLite for now. Instead of more prompting, we will soon write a wrapper that makes the SQLite API look more like what LLMs expect — this generally works better.
+
+### Model choice
+
+Claude Sonnet 3.5 is clearly the best model for writing code right now. However we found that it can be very deterministic, so it can help to crank up its `temperature`.
+
+We also played with `gpt-4o` vs `gpt-4o-mini`. The `mini` version is much cheaper but much worse, though it's pretty good at making websites, especially when given some examples.
+
+### Make real
+
+You might have seen the ["Make Real"](https://github.com/tldraw/make-real) demos of [tldraw](https://tldraw.dev/), where you can draw shapes on a canvas, and turn it into HTML. I made a prototype "Make Real with Backend". For this, I put tldraw in a [val](https://www.val.town/v/janpaul123/valledraw) (which you can fork), and put my VALL-E prototype in an iframe:
+
+<figure style={{ maxWidth: "600px" }}>
+  <video controls loop>
+    <source src="/video/make-real-with-backend.mp4" />
+  </video>
+  <figcaption>
+    "Make Real with Backend"
+  </figcaption>
+</figure>
+
+It currently only works with text prompts, but it should be easy to pass the SVGs of arbitrary drawings to the LLM, as with the original Make Draw demos.
+
+I modified the VALL-E prototype to pass the name of generated val up to tldraw, using the [`postMessage` API](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage). That way you can use a previously generated val as the basis for a new one, such as when I write "add more sample stories" in the video above.
+
+### Putting it all together
+
+Once I had built all this, I gave a lightning talk at an event hosted by [South Park Commons](https://www.southparkcommons.com/). It's 5 minutes of building lots of little apps.
+
+<iframe width="600" height="378" src="https://www.youtube.com/embed/XXMAwpOCiMI?si=RKFoGSxrFMcwq8cD" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
+
+# Reducing costs
+
+Since I started with a maximalist approach — put as much as possible in the context window — generation was costly and slow. Now let's look at ways to make it cheap and fast.
```

Comment:
I would talk more about the maximalist approach and how freaking many examples you shoved in there. I'd also spell out the pricing trade-off for folks. That's really interesting to me!

Reply:
You could also talk about how we had to get special anthropic rate limits lol